lib/mongo/error/invalid_replacement_document.rb

```diff
@@ -19,23 +19,20 @@ module Mongo
   class Error
 
     # Exception raised if the object is not a valid replacement document.
-    #
-    # @since 2.0.0
     class InvalidReplacementDocument < Error
 
       # The error message.
       #
-      # @since 2.0.0
+      # @deprecated
       MESSAGE = 'Invalid replacement document provided'.freeze
 
       # Instantiate the new exception.
       #
-      # @example Instantiate the exception.
-      #   Mongo::Error::InvalidReplacementDoc.new
-      #
-      # @since 2.0.0
-      def initialize
-        super(MESSAGE)
+      # @param [ String ] key The invalid key.
+      def initialize(key)
```

Comment:
This is an API change, the key should be defaulted to nil and my preference would be for it to be a keyword argument.

Reply:
done